common/reactive/src/test/java/io/helidon/common/reactive/BaseProcessorTest.java

```diff
@@ -237,6 +238,7 @@ public void testDoubleSubscribe() {
     }
 
     @Test
+    @Disabled("Against https://github.com/reactive-streams/reactive-streams-jvm#2.5")
     public void testSubscriptionNotCanceled() {
         TestProcessor<String> processor = new TestProcessor<>();
         TestSubscription subscription = new TestSubscription();
```

Comment:
@romain-grecourt It's kind of a complicated to find out if subscription was cancelled without making ugly wrapper, is there some more exact use-case behind this test, maybe we can find some more elegant solution?

Reply:
IIRC this is because `HttpRequestScopedPublisher` closes the connection when the subscription is canceled (see `hookOnCancel`).

This is not compliant with the reactive spec because I did not want to change this behavior and instead worked around the issue. This test simulates that context to make sure that we do not end-up closing connections (the test subscription represents the subscription to HttpRequestScopedPublisher).

I do not know the original reason behind this behavior, I'm guessing it was by design so that a user can close the underlying connection by canceling the subscription.

I'd say it's fine to tackle this issue for 2.0, i.e change `HttpRequestScopedPublisher.hookOnCancel` to not close the connection and remove the non compliant test + base processor implementation.

We can also discuss the need for a proper API to close the connection, maybe on `ServerRequest`.

@tomas-langer thoughts ?

Reply:
Removed connection close based on stream cancel as @romain-grecourt suggested in 5fa0005, I would create new issue for addition API change replacing this functionality to solve it separately if thats ok.

Reply:
Make sure you reference this PR from the issue to be created for a cross-reference.

Reply:
Issue #1278 created